x-pack/plugins/lists/server/services/exception_lists/create_endpoint_trusted_apps_list.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from 'kibana/server';
+import uuid from 'uuid';
+
+import {
+  ENDPOINT_TRUSTED_APPS_LIST_DESCRIPTION,
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+  ENDPOINT_TRUSTED_APPS_LIST_NAME,
+} from '../../../common/constants';
+import { ExceptionListSchema, ExceptionListSoSchema, Version } from '../../../common/schemas';
+
+import { getSavedObjectType, transformSavedObjectToExceptionList } from './utils';
+
+interface CreateEndpointListOptions {
+  savedObjectsClient: SavedObjectsClientContract;
+  user: string;
+  tieBreaker?: string;
+  version: Version;
+}
+
+/**
+ * Creates the Endpoint Trusted Apps agnostic list if it does not yet exist
+ *
+ * @param savedObjectsClient
+ * @param user
+ * @param tieBreaker
+ * @param version
+ */
+export const createEndpointTrustedAppsList = async ({
+  savedObjectsClient,
+  user,
+  tieBreaker,
+  version,
+}: CreateEndpointListOptions): Promise<ExceptionListSchema | null> => {
+  const savedObjectType = getSavedObjectType({ namespaceType: 'agnostic' });
+  const dateNow = new Date().toISOString();
+  try {
+    const savedObject = await savedObjectsClient.create<ExceptionListSoSchema>(
+      savedObjectType,
+      {
+        _tags: [],
+        comments: undefined,
+        created_at: dateNow,
+        created_by: user,
+        description: ENDPOINT_TRUSTED_APPS_LIST_DESCRIPTION,
+        entries: undefined,
+        immutable: false,
+        item_id: undefined,
+        list_id: ENDPOINT_TRUSTED_APPS_LIST_ID,
+        list_type: 'list',
+        meta: undefined,
+        name: ENDPOINT_TRUSTED_APPS_LIST_NAME,
+        tags: [],
+        tie_breaker_id: tieBreaker ?? uuid.v4(),
+        type: 'endpoint',
```

Comment:
Since this list has different restrictions I would use a different list type such as `endpointTrustedApps`

Reply:
Agreed, we should be thinking about this as a different `type` for at least 2 reasons that I can think of. 1) to ensure that it doesn't have any side effects on existing code / API consumers, 2) in case we want to switch on this for schema customizations later one (I realize we can switch on the id/name instead, but it might be a good idea to have both available since we know that future features will require multiple lists, e.g. per-policy trusted apps or per-policy exceptions).

Reply:
I thiiiiink the convention for IDs/types like this is hyphen-delimited though, as opposed to camelcase. So maybe something like `trusted-apps`. Could be wrong, and super nitpicky anyway, but maybe look for some examples before changing.

Reply:
Sounds good. I will create a new type (was trying to minimize the impact to manifest_manager, but its a minor refactoring there).